k eks nodes segfaults

I thought I did something wrong with an older version of this plugin but I just set up a brand new 1.27 cluster using karpenter and I get the same segfault output

```
k eks nodes
NAME     INSTANCE-TYPE     OS     CAPACITY-TYPE     REGION     AMI-ID     AMI-NAME     AGE
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x37fb074]

goroutine 1 [running]:
github.com/surajincloud/kubectl-eks/cmd.nodes(0x5aa6740?, {0x3f6b1fc?, 0x0?, 0x0?})
        /home/runner/work/kubectl-eks/kubectl-eks/cmd/nodes.go:67 +0x794
github.com/spf13/cobra.(*Command).execute(0x5aa6740, {0x5bafc18, 0x0, 0x0})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0x5aa6a20)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
github.com/surajincloud/kubectl-eks/cmd.Execute()
        /home/runner/work/kubectl-eks/kubectl-eks/cmd/root.go:25 +0x25
main.main()
        /home/runner/work/kubectl-eks/kubectl-eks/main.go:6 +0x17
```
I haven't dug into it to find out where it's happening but I thought I would open this ticket to track it in case you already know about the problem.

I don't have problems with getting the nodes regularly
```
k get nodes
NAME                                            STATUS   ROLES    AGE     VERSION
ip-192-168-100-101.us-west-2.compute.internal   Ready    <none>   7m30s   v1.27.1-eks-2f008fe
ip-192-168-132-133.us-west-2.compute.internal   Ready    <none>   7m28s   v1.27.1-eks-2f008fe
ip-192-168-158-101.us-west-2.compute.internal   Ready    <none>   107m    v1.27.1-eks-2f008fe
ip-192-168-172-71.us-west-2.compute.internal    Ready    <none>   107m    v1.27.1-eks-2f008fe
ip-192-168-78-24.us-west-2.compute.internal     Ready    <none>   7m28s   v1.27.1-eks-2f008fe
```

--- a/cmd/nodes.go
+++ b/cmd/nodes.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"log"
 	"os"
+	"strings"
 	"text/tabwriter"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
@@ -54,12 +55,28 @@ func nodes(cmd *cobra.Command, args []string) error {
 	defer w.Flush()
 	fmt.Fprintln(w, "NAME", "\t", "INSTANCE-TYPE", "\t", "OS", "\t", "CAPACITY-TYPE", "\t", "REGION", "\t", "AMI-ID", "\t", "AMI-NAME", "\t", "AGE")
 	for _, i := range nodeList {
+		var amiID, amiName, capacityType string
 		age := kube.GetAge(i.CreationTimestamp)
-		img := i.Labels[kube.NodeGroupImage]
-		dis, _ := ec2Client.DescribeImages(ctx, &ec2.DescribeImagesInput{ImageIds: []string{img}})
 
-		amiName := aws.ToString(dis.Images[0].Name)
-		fmt.Fprintln(w, i.Name, "\t", i.Labels[kube.InstanceTypeLabel], "\t", i.Labels[kube.OsLabel], "\t", i.Labels[kube.CapacityTypeLabel], "\t", i.Labels[kube.ZoneLabel], "\t", i.Labels[kube.NodeGroupImage], "\t", amiName, "\t", age)
+		// AMI
+		if i.Labels[kube.NodeGroupImage] == "" {
+			amiID = i.Labels[kube.KarpenterImage]
+			dis, _ := ec2Client.DescribeImages(ctx, &ec2.DescribeImagesInput{ImageIds: []string{amiID}})
+			amiName = aws.ToString(dis.Images[0].Name)
+		} else {
+			amiID = i.Labels[kube.NodeGroupImage]
+			dis, _ := ec2Client.DescribeImages(ctx, &ec2.DescribeImagesInput{ImageIds: []string{amiID}})
+			amiName = aws.ToString(dis.Images[0].Name)
+		}
+
+		// Capacity Type
+		if i.Labels[kube.CapacityTypeLabel] != "" {
+			capacityType = i.Labels[kube.CapacityTypeLabel]
+		} else {
+			capacityType = strings.ToUpper(i.Labels[kube.KarpenterCapacityTypeLabel])
+
+		}
+		fmt.Fprintln(w, i.Name, "\t", i.Labels[kube.InstanceTypeLabel], "\t", i.Labels[kube.OsLabel], "\t", capacityType, "\t", i.Labels[kube.ZoneLabel], "\t", amiID, "\t", amiName, "\t", age)
 	}
 	return nil
 }

--- a/pkg/kube/labels.go
+++ b/pkg/kube/labels.go
@@ -1,15 +1,21 @@
 package kube
 
 const (
-	CapacityTypeLabel = "eks.amazonaws.com/capacityType"
-	NodeGroupLabel    = "eks.amazonaws.com/nodegroup"
-	ComputeType       = "eks.amazonaws.com/compute-type" // to detect fargate nodes
-	ArchLabel         = "kubernetes.io/arch"
-	OsLabel           = "kubernetes.io/os"
-	HostNameLabel     = "kubernetes.io/hostname"
+	// capacity Type
+	CapacityTypeLabel          = "eks.amazonaws.com/capacityType"
+	KarpenterCapacityTypeLabel = "karpenter.sh/capacity-type"
+
+	NodeGroupLabel = "eks.amazonaws.com/nodegroup"
+	ComputeType    = "eks.amazonaws.com/compute-type" // to detect fargate nodes
+	ArchLabel      = "kubernetes.io/arch"
+	OsLabel        = "kubernetes.io/os"
+	HostNameLabel  = "kubernetes.io/hostname"
 
 	InstanceTypeLabel = "node.kubernetes.io/instance-type"
 
-	ZoneLabel      = "topology.kubernetes.io/zone"
+	ZoneLabel = "topology.kubernetes.io/zone"
+
+	// Ami ID
 	NodeGroupImage = "eks.amazonaws.com/nodegroup-image"
+	KarpenterImage = "karpenter.k8s.aws/instance-ami-id"
 )

